src/utils/updateTodo.ts

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable @typescript-eslint/indent */
+import { ErrorMessages } from '../enums';
+import { Todo } from '../types/Todo';
+
+export const updateTodos = (
+  todoIds: number[],
+  newStatus: boolean | null,
+  setLoadingTodosCount: React.Dispatch<React.SetStateAction<number[]>>,
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  setErrorMessage: React.Dispatch<React.SetStateAction<ErrorMessages>>,
+  operation: (id: number, changes?: Partial<Todo>) => Promise<void>,
+  deleteOperation?: (id: number) => Promise<void>,
+): Promise<void> => {
+  setLoadingTodosCount(current => [...current, ...todoIds]);
+
+  const promises = todoIds.map(id => {
+    const action =
+      newStatus !== null
+        ? operation(id, { completed: newStatus })
+        : deleteOperation
+          ? deleteOperation(id)
+          : operation(id);
```

Comment:
The nested ternary operator can be hard to read. Consider refactoring that section for clarity.